server/lib/nl/utils.py

```diff
@@ -587,9 +587,16 @@ def place_detection_with_heuristics(query_fn, query: str) -> List[str]:
     # now find "united states" as a place. Therefore, to avoid such situations
     # we should try to find the place string found in the original (lower case)
     # query string.
-    if not ignore and places_found[i] in query_lower:
+    # If places_found[i] was a special place, keep it always.
+    if (places_found[i]
+        in constants.SPECIAL_PLACES) or (not ignore and
+                                         places_found[i] in query_lower):
       places_to_return.append(places_found[i])
 
+  if 'oceania' in query_lower:
```

Comment:
Meant to keep it? use logging.info() so line numbers may show up?

Reply:
removed